src/openqaoa-core/tests/test_workflows.py

```diff
@@ -1726,6 +1731,640 @@ def test_change_properties_after_compilation(self):
+            for res, res4 in zip(result, result4):
+                self.assertAlmostEqual(
+                    res,
+                    res4,
+                    places=15,
+                    msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should return the same result \
+                    when passing the optimized params manually",
+                )
+
+            # evaluate the circuit with a wrong input, it should raise an error
+            with self.assertRaises(
+                TypeError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                  passing a wrong input"
+            ):
+                fqaoa.evaluate_circuit(1)
+
+            # evaluate the circuit with a list longer than it should, it should raise an error
+            with self.assertRaises(
+                AssertionError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                passing a list longer than it should"
+            ):
+                fqaoa.evaluate_circuit(params2 + [1])
+
+            # evaluate the circuit with a list shorter than it should, it should raise an error
+            with self.assertRaises(
+                AssertionError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                passing a list shorter than it should"
+            ):
+                fqaoa.evaluate_circuit(params2[:-1])
+
+            # evaluate the circuit with a dict with a wrong key, it should raise an error
+            with self.assertRaises(
+                KeyError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error \
+                when passing a dict with a wrong key"
+            ):
+                fqaoa.evaluate_circuit({**params, "wrong_key": 1})
+
+            # evaluate the circuit with a dict with a value longer than it should, it should raise an error
+            with self.assertRaises(
+                ValueError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                passing a dict with a value longer than it should"
+                ):
+                fqaoa.evaluate_circuit(
+                    {**params, list(params.keys())[0]: np.random.rand(40)}
+                )
+
+            # evaluate the circuit without passing any param, it should raise an error
+            with self.assertRaises(
+                TypeError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                    not passing any param",
+                ):
+                fqaoa.evaluate_circuit()
+
+    def test_qaoa_evaluate_circuit_shot(self):
```

Comment:
Let's rename this to `test_fqaoa_evaluate_circuit_shot`

Reply:
Rename `test_fqaoa_evaluate_circuit_shot`.